pkg/planner/core/indexmerge_path.go

```diff
@@ -596,7 +930,183 @@ func (ds *DataSource) generateIndexMergeOnDNF4MVIndex(normalPathCnt int, filters
+				IndexRangeScan(a, [3,3])              --- COP
+			IndexRangeScan(non-mv-index-if-any)(?)    --- COP
+			TableRowIdScan(t)                         --- COP
+	3. select * from t where ((1 member of (a) and c=1) or (json_overlap(a, '[1, 2,3]') and d=2) or (other index predicates)
+		analyze: find and utilize index access filter items as much as possible:
+		IndexMerge(OR-UNION)                          --- ROOT
+			IndexRangeScan(mv-index-a)(1)             --- COP
+			IndexMerge(mv-index-b OR-UNION)           --- ROOT (embedded index merge) ---> simplify: we can merge with outer index union merge, and defer TableRowIdScan(t).
+				IndexRangeScan(a, [1,1])              --- COP
+				IndexRangeScan(a, [2,2])              --- COP
+				IndexRangeScan(a, [3,3])              --- COP
+			IndexRangeScan(non-mv-index-if-any)(?)    --- COP
+			TableRowIdScan(t)                         --- COP
+*/
+func (ds *DataSource) generateIndexMerge4ComposedIndex(normalPathCnt int, indexMergeConds []expression.Expression) error {
+	isPossibleIdxMerge := len(indexMergeConds) > 0 && // have corresponding access conditions, and
+		len(ds.possibleAccessPaths) > 1 // have multiple index paths
+	if !isPossibleIdxMerge {
+		return nil
+	}
+
+	if len(indexMergeConds) == 1 {
+		// DNF path.
+		sf, ok := indexMergeConds[0].(*expression.ScalarFunction)
+		if !ok || sf.FuncName.L != ast.LogicOr {
+			// targeting: cond1 or cond2 or cond3
+			return nil
+		}
+		dnfFilters := expression.FlattenDNFConditions(sf)
+		mvIndexPartialPaths, usedAccessMap, needSelection4MVIndex, err := ds.generateIndexMergePartialPath4Or(normalPathCnt, dnfFilters)
+		if err != nil {
+			return err
+		}
+		if len(mvIndexPartialPaths) == 0 {
+			// no mv index partial paths to be composed of.
+			return nil
+		}
+		normalIndexPartialPaths, usedAccessMap, needSelection4NormalIndex, err := ds.generateNormalIndexPartialPath4Or(dnfFilters, usedAccessMap, normalPathCnt)
+		if err != nil {
+			return err
+		}
+		// since multi normal index merge path is handled before, here focus on multi mv index merge, or mv and normal mixed index merge
+		composed := (len(mvIndexPartialPaths) > 1) || (len(mvIndexPartialPaths) == 1 && len(normalIndexPartialPaths) >= 1)
+		if !composed {
+			return nil
+		}
+		// if any cnf item is not used as index partial path, index merge is not valid anymore.
+		for _, one := range usedAccessMap {
+			if !one {
+				return nil
+			}
+		}
+		// todo: make this code as a portal of all index merge path.
+		// if we derive:
+		// 1: some mv index partial path, no normal index path, it means multi mv index merge.
+		// 2: some mv index partial path, some normal index path, it means hybrid index merge.
+		// 3: no mv index partial path, several normal index path, it means multi normal index merge.
+		var combinedPartialPaths []*util.AccessPath
+		combinedPartialPaths = normalIndexPartialPaths
+		combinedPartialPaths = append(combinedPartialPaths, mvIndexPartialPaths...)
```

Comment:
If you want to reuse the slice, this can be more simplified.
```suggestion
		combinedPartialPaths := append(normalIndexPartialPaths, mvIndexPartialPaths...)
```